netplugin/netd.go

```diff
@@ -485,24 +494,68 @@ func handleEvents(netPlugin *plugin.NetPlugin, crt *crt.Crt, opts cliOpts) error
 
 	err := <-recvErr
 	if err != nil {
-		log.Printf("Failure occured. Error: %s", err)
+		log.Errorf("Failure occured. Error: %s", err)
 		return err
 	}
 
 	return nil
 }
 
+func configureSyslog(syslogParam string) {
+	var err error
+	var hook log.Hook
+
+	// disable colors if we're writing to syslog *and* we're the default text
+	// formatter, because the tty detection is useless here.
+	if tf, ok := log.StandardLogger().Formatter.(*log.TextFormatter); ok {
+		tf.DisableColors = true
+	}
+
+	if syslogParam == "kernel" {
+		hook, err = logrus_syslog.NewSyslogHook("", "", syslog.LOG_INFO, "netplugin")
+		if err != nil {
+			log.Fatalf("Could not connect to kernel syslog")
+			os.Exit(1)
```

Comment:
I have the same question, doing explicit exit after Fatalf would seem like Fatalf may not panic/exit.